core/common/src/main/java/alluxio/underfs/UfsFileStatus.java

```diff
@@ -24,6 +24,9 @@
   public static final String INVALID_CONTENT_HASH = "";
   public static final long UNKNOWN_BLOCK_SIZE = -1;
 
+  // if block size equals to MAGIC_BLOCK_SIZE, use alluxio.user.block.size.bytes.default instread
+  public static final long MAGIC_BLOCK_SIZE = 1;
```

Comment:
Since `1` is technically a valid number of bytes that someone could want to configure, what if we use `-1` or a negative value instead? It wouldn't make sense in any other case to configure a negative value, so I feel the magic number should be at least `<= 0`

Reply:
please fine code in [S3AFileSystem](https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java), `-1` is invalid number for s3 filesystem, the minimal number is `1`.

Another option is `Long.MAX_VALUE`, but still have same issue as you mentioned.

Reply:
@ZacBlanco any suggestions?